File: ruby/v1_20/get_dfa_sites.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Author:: api.jimper@gmail.com (Jonathon Imperiosi)
#
# Copyright:: Copyright 2014, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example gets existing DFA sites based on a given search criteria.
# Results are limited to the first 10.
#
# Tags: site.getDfaSites

require_relative 'dfa_utils'

def get_dfa_sites(search_string, auth_token)
  # Get the site service.
  site_service = Savon.client(
    :wsdl => DfaUtils.generate_wsdl_url('site'),
    # Set the WSSE authentication header.
    :wsse_auth => [DfaUtils.get_dfa_username(), auth_token])

  # Create DFA site search criteria structure.
  dfa_site_search_criteria = {
    :page_number => 0,
    :page_size => 10,
    :search_string => search_string
  }

  # Fetch the DFA sites.
  response = site_service.call(
    :get_dfa_sites,
    :message => {:dfa_site_search_criteria => dfa_site_search_criteria},
    :soap_header => DfaUtils.generate_request_header())

  response = response.to_hash[:multi_ref]

  # Display DFA site names and IDs.
  if response[0] && response[0][:records]
    record_set = response[0][:records][:records]
    # Ensure records are wrapped in an array, even if there is only one
    # record.
    record_set = [record_set] unless record_set.is_a?(Array)

    record_set.each do |record|
      # Extract the href from the record. The href will specify which element
      # of the response multi_ref array contains the object this record
      # refers to.
      record_href = record[:@href][/#id(.*)/,1].to_i
      dfa_site = response[record_href]
      puts 'DFA site with name "%s" and ID %d was found.' %
          [dfa_site[:name], dfa_site[:id]]
    end
  else
    puts 'No DFA sites found for your criteria.'
  end
end

if __FILE__ == $0
  # Provide criteria to search upon.
  search_string = 'INSERT_SEARCH_STRING_CRITERIA_HERE'

  # Generate a DFA authentication token.
  auth_token = DfaUtils.authenticate()

  get_dfa_sites(search_string, auth_token)
end

File: php/v1.20/GetDfaSites.php
<?php
/**
 * This example gets existing DFA sites based on a given search criteria.
 * Results are limited to the first 10.
 *
 * Tags: sites.getDfaSites
 *
 * PHP version 5.3
 * PHP extensions: google/apiclient, SoapClient.
 *
 * Copyright 2014, Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * @package    GoogleApiAdsDfa
 * @category   WebServices
 * @copyright  2014, Google Inc. All Rights Reserved.
 * @license    http://www.apache.org/licenses/LICENSE-2.0 Apache License,
 *             Version 2.0
 * @author     Jonathon Imperiosi <api.jimper@gmail.com>
 */

require_once 'DfaUtils.php';

// Provide criteria to search upon.
$searchString = 'INSERT_SEARCH_STRING_CRITERIA_HERE';
$networkId = (float) 'INSERT_NETWORK_ID_HERE';
$subnetworkId = (float) 'INSERT_SUBNETWORK_ID_HERE';

// Set SOAP and XML settings.
$siteWsdl = 'https://advertisersapi.doubleclick.net/v1.20/api/' .
    'dfa-api/site?wsdl';
$options = array('encoding' => 'utf-8');

// Authenticate with the API and retrive auth token
$token = DfaUtils::authenticate();

// Get SiteService.
$siteService = new SoapClient($siteWsdl, $options);

// Set headers.
$headers = array(DfaUtils::generateWsseHeader($token),
    DfaUtils::generateRequestHeader());
$siteService->__setSoapHeaders($headers);

// Create site search criteria structure.
$siteSearchCriteria = array(
    'pageNumber' => 0,
    'pageSize' => 10,
    'searchString' => $searchString,
    'excludeSitesMappedToSiteDirectory' => false,
    'networkId' => $networkId,
    'subnetworkId' => $subnetworkId);

try {
  // Fetch the DFA sites.
  $result = $siteService->getDfaSites($siteSearchCriteria);
} catch (Exception $e) {
  print $e->getMessage();
  exit(1);
}

// Display DFA site names and IDs.
if (isset($result->records)) {
  foreach($result->records as $dfaSite) {
    print 'DFA site with name "' . $dfaSite->name . '" and ID '
        . $dfaSite->id . " was found.\n";
  }
} else {
  print "No DFA sites found for your criteria.\n";
}
